core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/collection/SatisfiesExpression.kt

```diff
@@ -27,13 +29,13 @@ enum class SatisfiesType {
 
 sealed class SatisfiesExpression<T : ValidType>(
     private val satisfiesType: SatisfiesType,
-    private val list: TypeExpression<ArrayType<T>>,
-    private val variable: String,
+    private val arrayExpression: TypeExpression<ArrayType<T>>,
+    private val iteratorName: String,
     private val predicate: (Iterator<T>) -> TypeExpression<BooleanType>,
 ) : TypeExpression<BooleanType> {
     override fun toDopeQuery(): DopeQuery {
-        val listDopeQuery = list.toDopeQuery()
-        val iteratorVariable = if (variable == DEFAULT_ITERATOR_VARIABLE) variable + IteratorManager.count else variable
+        val listDopeQuery = arrayExpression.toDopeQuery()
+        val iteratorVariable = if (iteratorName == DEFAULT_ITERATOR_VARIABLE) IteratorManager.getIteratorName() else iteratorName
```

Comment:
This is a but stupid, you misuse the constant DEFAULT_ITERATOR_VARIABLE here just to check if the user entered an iteratorName. As dicussed offline, just make it nullable

Reply:
done